draft-ietf-httpbis-connect-tcp.md

```diff
@@ -66,7 +66,23 @@ This specification describes an alternative mechanism for proxying TCP in HTTP.
 
 A template-driven TCP transport proxy for HTTP is identified by a URI Template {{!RFC6570}} containing variables named "target_host" and "tcp_port".  The client substitutes the destination host and port number into these variables to produce the request URI.
 
-The "target_host" variable MUST be a domain name, an IP address literal, or a list of IP addresses.  The "tcp_port" variable MUST be a single integer.  If "target_host" is a list (as in {{Section 3.2.1 of !RFC6570}}), the server SHOULD perform the same connection procedure as if these IP addresses had been returned in response to A and AAAA queries for a domain name.
+The "target_host" variable MUST be a domain name, an IP address literal, or a list of IP addresses.  If "target_host" is a list (as in {{Section 3.2.1 of !RFC6570}}), the server SHOULD perform the same connection procedure as if these IP addresses had been returned in response to A and AAAA queries for a domain name.
+
+The "tcp_port" variable MUST represent a single integer between 1 and 65535 inclusive.  Both the "target_host" and "tcp_port" values MUST NOT be empty or undefined when using the template for TCP proxying.
+
+Using the terms `IPv6address`, `IPv4address`, `reg-name`, and `port` from {{!RFC3986}} and notation from {{!RFC5234}} and {{!RFC6570}}, the "target_host" and "tcp_port" values MUST adhere to the format in {{abnf}}:
+
+~~~~ abnf
+ip-addr = IPv6address / IPv4address
+single-host-str = DQUOTE ( ip-addr / reg-name ) DQUOTE
+ip-addr-str = DQUOTE ip-addr DQUOTE
+ip-addr-list = "(" ( ip-addr-str ", " )* ip-addr-str ")"
```

Comment:
This ABNF doesn't match the example, it includes DQUOTES and parentheses that aren't in the example. Why not simply reuse the syntax from 9298 with just the tweak for the list?

Reply:
Yes, the quotes and parentheses are from RFC 6570's value syntax.  They're needed in order to distinguish lists from strings, since ABNF has no concept of a list.

It sounds like you'd prefer to use ABNF only for the individual list elements, and leave the description of the list itself in English?

Reply:
I guess I'm confused, are the double quotes and parentheses required? If so, can you add them to the examples so I can visualize the syntax better?

Reply:
I'm even more confused by 5f5e1ac8645bd8352bbd1c372bf6178782d3e538. I assumed that the ABNF covered what would end up in the path, not in the "Variable assignments in RFC 6570 syntax". What am I missing?

Reply:
I agree that it's confusing.  ABNF is not really able to do what we need here.  I've moved more of the description to English to reduce the complexity.

Reply:
Why not? Doesn't this work?
```
ip-addr = IPv6address / IPv4address
ip-addr-list = *( ip-addr "," ) ip-addr
target_host = reg-name / ip-addr-list
```

Reply:
That's an option, but it would result in a change in the behavior because it would not respect the explode modifier.  I've added more examples that I hope make this clear.

RFC 6570 does not really define a canonical string representation for lists, so we can't specify them using ABNF.

Reply:
I see. If it were up to me, I would remove the explode modifier and only use the ABNF - that way we only have a single way to convey the information, and it's formally defined.